src/libstd/sys/unix/thread.rs

```diff
@@ -314,21 +316,30 @@ pub fn sleep(dur: Duration) {
 // is created in an application with big thread-local storage requirements.
 // See #6233 for rationale and details.
 //
-// Link weakly to the symbol for compatibility with older versions of glibc.
-// Assumes that we've been dynamically linked to libpthread but that is
-// currently always the case.  Note that you need to check that the symbol
-// is non-null before calling it!
+// Use dlsym to get the symbol value at runtime, for compatibility
```

Comment:
compatibility is not the reason anymore - this is done to avoid the debian issue, regardless of the glibc version